owners/src/owners.js

```diff
@@ -115,16 +128,16 @@ class OwnersTree {
   }
 
   /**
-   * Tests if a user is in the ownership path of the tree.
+   * Tests if a user is in the ownership path of a file.
    *
+   * @param {!string} filename file to test ownership for.
    * @param {!string} username user to check ownership of.
-   * @return {boolean} true of the username is in this or an ancestor's OWNERS.
+   * @return {boolean} true of the user is an owner of the file.
    */
-  hasOwner(username) {
-    const allOwners = this.allRules
-      .map(rule => rule.owners)
-      .reduce((left, right) => left.concat(right), []);
-    return allOwners.includes(username);
+  fileHasOwner(filename, username) {
+    const allRules = this.atPath(filename).allRules;
+    const fileRules = allRules.filter(rule => rule.matchesFile(filename));
+    return fileRules.some(rule => rule.owners.includes(username));
```

Comment:
You can merge these two lines together:
`return allRules.some(rule => rule.matchesFile(filename) && rule.owners.includes(username));`
This will short-circuit with a `true` value on the first rule that matches

Reply:
I'm aware this could be one-lined, I find that it's more readable to break it into steps this way (and there's a very high likelihood that Closure+V8 optimize the **** out of map/filter/reduce functions anyway)

Reply:
I don't think it's unreadable to have these two together - basically it says "at least one rule both matches the file and is owned by the user - it's a pretty simple rule :)